public/app/controllers/mainCtrl.js

```diff
@@ -40,63 +37,25 @@ angular.module('kibibitCodeEditor')
         controller: 'projectFolderModalController',
         controllerAs: 'projectFolderModalCtrl',
         className: 'ngdialog-theme-default',
-        scope: $scope
+        scope: $scope,
+        data: {
+          userHomeDirectoryPath: vm.userHomeDirectoryPath
+        }
+      }).closePromise.then(function(selectedProjectFolder) {
+        EventManagerService.trigger(
```

Comment:
I removed the eventManager injection and used the main ctrl as a "bridge" to pass the path back to the tree